### PR TITLE
Feature/smar92 availability

### PR DESCRIPTION
**Pull request by SwiftXLight**
Description changes:
• Add ability to create working hours
• Add ability to delete working hours
• Add ability to change timezone
• Add validation for input data

### General
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Don't have "any" on my code
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] All links in on ENV files
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

#### General view:
image

#### Modal for creating working hours:
image

#### Errors example:
image
image

#### Successful creation:
image
image

#### Deletion element:
image
image

#### Changing timezone:
image
image
image

### CODE REVIEW

**Comment by lyudad** [comment]
why we need "select timezome" in this screen? we can detect it automatecally

File: src/components/Scheduler/Modals/PopupCreateContent.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                            value={selectedRange?.end ?? ''}
+                            onChange={handleEndChange}
+                        />
+                    </WrapperLabelAndInput>
+                    <ModalButtonsWrapper>
+                        <CancelButton
+                            onClick={() => {
+                                setShowCreatePopup(false);
+                                setErrorMessage('');
+                            }}
```
Comment (lyudad) [inline src/components/Scheduler/Modals/PopupCreateContent.tsx]: why its in render, not in separate function?

File: src/components/Scheduler/Modals/styles.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+`;
+
+export const Label = styled.label`
+`;
+
+export const DateInput = styled.input`
+`;
+
+export const SelectedDateText = styled.p`
+`;
```
Comment (lyudad) [inline src/components/Scheduler/Modals/styles.ts]: why line 65 - 72 needed?

Comment (SwiftXLight) [reply src/components/Scheduler/Modals/styles.ts]: deleted

File: src/components/Scheduler/Modals/PopupCreateContent.tsx (hunk 1 above)
Comment (SwiftXLight) [reply src/components/Scheduler/Modals/PopupCreateContent.tsx]: moved to separate function

**Comment by SwiftXLight** [comment]
> why we need "select timezome" in this screen? we can detect it automatecally

Yes and it is already detect it automatically, but this select timezone gives doctor ability to see how his/her working hours looks in different timezones. But creating work hours still will be done in local time. In our team project design we have that. Is that functionality unnecessary or leave it?

File: src/components/Scheduler/index.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+import PopupDeleteContent from './Modals/PopupDeleteContent';
+import PopupCreateContent from './Modals/PopupCreateContent';
+import TimezoneSelect from './TimezoneSelect/TimezoneSelect';
+
+const defaultTZ = moment.tz.guess();
+
+export interface ISelectedRange {
+    start: null | number | string;
+    end: null | number | string;
+};
```
Comment (lyudad) [inline src/components/Scheduler/index.tsx:19]: here should be empty line

File: src/components/Scheduler/index.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    uuid: string;
+    title: string;
+    start: Date;
+    end: Date;
+};
+  
+function Scheduler() {
+    const { t } = useTranslation();
+
+    const [showWarning, setShowWarning] = useState(false);
```
Comment (lyudad) [inline src/components/Scheduler/index.tsx]: where are types?

File: src/components/Scheduler/index.tsx (hunk 3 above)
Comment (SwiftXLight) [reply src/components/Scheduler/index.tsx:19]: add empty line

File: src/components/Scheduler/index.tsx (hunk 4 above)
Comment (SwiftXLight) [reply src/components/Scheduler/index.tsx]: add TFunction type

Comment (lyudad) [reply src/components/Scheduler/index.tsx]: where?

Comment (lyudad) [reply src/components/Scheduler/index.tsx]: line 31 - 33

Comment (SwiftXLight) [reply src/components/Scheduler/index.tsx]: add types for 31-33. In previos message I said, that I add typization for t function

**Pull request merged**